### ISSUE TITLE
Error running command, check - BCFtools error?

### ISSUE THREAD

**Issue by sgl07007** [tseemann/snippy#353]
Hello, 

I am running snippy 4.4.5. Most of the time everything works perfectly, but for a few samples I get the following output/error: 

snippy --outdir JKS2176_snippy_reads_chromosome --ref JKS2530_chrom.gbk --R1 JKS2176_TruSeq_output_R1_paired.fastq --R2 JKS2176_TruSeq_output_R2_paired.fastq --report --unmapped

[11:55:11] This is snippy 4.4.5
[11:55:11] Written by Torsten Seemann
[11:55:11] Obtained from https://github.com/tseemann/snippy
[11:55:11] Detected operating system: linux
[11:55:11] Enabling bundled linux tools.
[11:55:11] Found bwa - /isg/shared/apps/snippy/4.4.5/bin/bwa
[11:55:11] Found bcftools - /isg/shared/apps/snippy/4.4.5/bin/bcftools
[11:55:11] Found samtools - /isg/shared/apps/snippy/4.4.5/bin/samtools
[11:55:11] Found java - /isg/shared/apps/snippy/4.4.5/bin/java
[11:55:11] Found snpEff - /isg/shared/apps/snippy/4.4.5/bin/snpEff
[11:55:11] Found samclip - /isg/shared/apps/snippy/4.4.5/bin/samclip
[11:55:11] Found seqtk - /isg/shared/apps/snippy/4.4.5/bin/seqtk
[11:55:11] Found parallel - /isg/shared/apps/snippy/4.4.5/bin/parallel
[11:55:11] Found freebayes - /isg/shared/apps/snippy/4.4.5/bin/freebayes
[11:55:11] Found freebayes-parallel - /isg/shared/apps/snippy/4.4.5/bin/freebayes-parallel
[11:55:11] Found fasta_generate_regions.py - /isg/shared/apps/snippy/4.4.5/bin/fasta_generate_regions.py
[11:55:11] Found vcfstreamsort - /isg/shared/apps/snippy/4.4.5/bin/vcfstreamsort
[11:55:11] Found vcfuniq - /isg/shared/apps/snippy/4.4.5/bin/vcfuniq
[11:55:11] Found vcffirstheader - /isg/shared/apps/snippy/4.4.5/bin/vcffirstheader
[11:55:11] Found gzip - /usr/bin/gzip
[11:55:11] Found vt - /isg/shared/apps/snippy/4.4.5/bin/vt
[11:55:11] Found snippy-vcf_to_tab - /isg/shared/apps/snippy/4.4.5/bin/snippy-vcf_to_tab
[11:55:11] Found snippy-vcf_report - /isg/shared/apps/snippy/4.4.5/bin/snippy-vcf_report
[11:55:11] Checking version: samtools --version is >= 1.7 - ok, have 1.9
[11:55:11] Checking version: bcftools --version is >= 1.7 - ok, have 1.9
[11:55:11] Checking version: freebayes --version is >= 1.1 - ok, have 1.3
[11:55:12] Checking version: java -version is >= 1.8 - ok, have 1.8
[11:55:12] Checking version: snpEff -version is >= 4.3 - ok, have 4.3
[11:55:12] Checking version: bwa is >= 7.12 - ok, have 7.17
[11:55:12] Using reference: /home/CAM/sgoldstein/Paper_3/GATK/JKS2530_chrom.gbk
[11:55:12] Treating reference as 'genbank' format.
[11:55:12] Will use 8 CPU cores.
[11:55:12] Using read file: /home/CAM/sgoldstein/Paper_3/GATK/JKS2176_TruSeq_output_R1_paired.fastq
[11:55:12] Using read file: /home/CAM/sgoldstein/Paper_3/GATK/JKS2176_TruSeq_output_R2_paired.fastq
[11:55:12] Creating folder: JKS2176_snippy_reads_chromosome
[11:55:12] Changing working directory: JKS2176_snippy_reads_chromosome
[11:55:12] Creating reference folder: reference
[11:55:12] Extracting FASTA and GFF from reference.
[11:55:18] Wrote 1 sequences to ref.fa
[11:55:18] Wrote 9804 features to ref.gff
[11:55:18] Creating reference/snpeff.config
[11:55:18] Freebayes will process 15 chunks of 408455 bp, 8 chunks at a time.
[11:55:18] Using BAM RG (Read Group) ID: JKS2176_snippy_reads_chromosome
[11:55:18] Running: samtools faidx reference/ref.fa 2>> snps.log
[11:55:18] Running: bwa index reference/ref.fa 2>> snps.log
[11:55:24] Running: mkdir -p reference/genomes && cp -f reference/ref.fa reference/genomes/ref.fa 2>> snps.log
[11:55:24] Running: ln -sf reference/ref.fa . 2>> snps.log
[11:55:24] Running: ln -sf reference/ref.fa.fai . 2>> snps.log
[11:55:24] Running: mkdir -p reference/ref && gzip -c reference/ref.gff > reference/ref/genes.gff.gz 2>> snps.log
[11:55:24] Running: snpEff build -c reference/snpeff.config -dataDir . -gff3 ref 2>> snps.log
[11:55:31] Running: bwa mem  -Y -M -R '@RG\tID:JKS2176_snippy_reads_chromosome\tSM:JKS2176_snippy_reads_chromosome' -t 8 reference/ref.fa /home/CAM/sgoldstein/Paper_3/GATK/JKS2176_TruSeq_output_R1_paired.fast
q /home/CAM/sgoldstein/Paper_3/GATK/JKS2176_TruSeq_output_R2_paired.fastq | samclip --max 10 --ref reference/ref.fa.fai | samtools sort -n -l 0 -T /tmp --threads 7 -m 571M | samtools fixmate -m - - | samtools
 sort -l 0 -T /tmp --threads 7 -m 571M | samtools markdup -T /tmp -r -s - - > snps.bam 2>> snps.log
[M::bwa_idx_load_from_disk] read 0 ALT contigs
[samclip] samclip 0.2 by Torsten Seemann (@torstenseemann)
[samclip] Loading: reference/ref.fa.fai
[samclip] Found 1 sequences in reference/ref.fa.fai
[M::process] read 539246 sequences (80000087 bp)...
[M::process] read 537702 sequences (80000364 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (61, 228758, 11, 10)
[M::mem_pestat] analyzing insert size distribution for orientation FF...
[M::mem_pestat] (25, 50, 75) percentile: (243, 389, 567)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1215)
[M::mem_pestat] mean and std.dev: (387.36, 218.62)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 1539)
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (452, 540, 638)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (80, 1010)
[M::mem_pestat] mean and std.dev: (549.15, 137.63)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 1196)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (418, 986, 4296)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 12052)
[M::mem_pestat] mean and std.dev: (2106.00, 1929.72)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 15930)
[M::mem_pestat] analyzing insert size distribution for orientation RR...
[M::mem_pestat] (25, 50, 75) percentile: (548, 1129, 1682)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3950)
[M::mem_pestat] mean and std.dev: (1046.50, 571.63)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 5084)
[M::mem_pestat] skip orientation FF
[M::mem_pestat] skip orientation RF
[M::mem_pestat] skip orientation RR
[M::mem_process_seqs] Processed 539246 reads in 113.682 CPU sec, 14.375 real sec
[samclip] Processed 100000 records...
[samclip] Processed 200000 records...
[samclip] Processed 300000 records...
[samclip] Processed 400000 records...
[samclip] Processed 500000 records...
[M::process] read 293206 sequences (41854620 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (76, 227657, 3, 12)
[M::mem_pestat] analyzing insert size distribution for orientation FF...
[M::mem_pestat] (25, 50, 75) percentile: (251, 365, 524)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1070)
[M::mem_pestat] mean and std.dev: (371.56, 186.88)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 1343)
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (455, 544, 643)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (79, 1019)
[M::mem_pestat] mean and std.dev: (553.20, 139.20)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 1207)
[M::mem_pestat] skip orientation RF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation RR...
[M::mem_pestat] (25, 50, 75) percentile: (568, 1703, 5552)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 15520)
[M::mem_pestat] mean and std.dev: (2856.08, 2449.58)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 20504)
[M::mem_pestat] skip orientation FF
[M::mem_pestat] skip orientation RR
[M::mem_process_seqs] Processed 537702 reads in 114.396 CPU sec, 14.331 real sec
[samclip] Processed 600000 records...
[samclip] Processed 700000 records...
[samclip] Processed 800000 records...
[samclip] Processed 900000 records...
[samclip] Processed 1000000 records...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (49, 123429, 1, 1)
[M::mem_pestat] analyzing insert size distribution for orientation FF...
[M::mem_pestat] (25, 50, 75) percentile: (209, 339, 527)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1163)
[M::mem_pestat] mean and std.dev: (356.30, 189.47)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 1481)
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (459, 547, 648)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (81, 1026)
[M::mem_pestat] mean and std.dev: (557.21, 140.71)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 1215)
[M::mem_pestat] skip orientation RF as there are not enough pairs
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_pestat] skip orientation FF
[M::mem_process_seqs] Processed 293206 reads in 65.243 CPU sec, 7.987 real sec
[samclip] Processed 1100000 records...
[samclip] Processed 1200000 records...
[samclip] Processed 1300000 records...
[main] Version: 0.7.17-r1188
[main] CMD: bwa mem -Y -M -R @RG\tID:JKS2176_snippy_reads_chromosome\tSM:JKS2176_snippy_reads_chromosome -t 8 reference/ref.fa /home/CAM/sgoldstein/Paper_3/GATK/JKS2176_TruSeq_output_R1_paired.fastq /home/CAM
/sgoldstein/Paper_3/GATK/JKS2176_TruSeq_output_R2_paired.fastq
[main] Real time: 40.336 sec; CPU: 295.372 sec
[samclip] Total SAM records 1376388, removed 42628, allowed 52031, passed 1333760
[samclip] Header contained 3 lines
[samclip] Done.
[bam_sort_core] merging from 0 files and 7 in-memory blocks...
[bam_sort_core] merging from 0 files and 7 in-memory blocks...
[11:56:50] Running: samtools index snps.bam 2>> snps.log
[11:56:52] Running: fasta_generate_regions.py reference/ref.fa.fai 408455 > reference/ref.txt 2>> snps.log
[11:56:52] Running: freebayes-parallel reference/ref.txt 8 -p 2 -P 0 -C 10 --min-repeat-entropy 1.5 --strict-vcf -q 13 -m 60 --min-coverage 10 -F 0.05  -f reference/ref.fa snps.bam > snps.raw.vcf 2>> snps.log
[11:57:28] Running: bcftools view --include 'FMT/GT="1/1" && QUAL>=100 && FMT/DP>=10 && (FMT/AO)/(FMT/DP)>=0' snps.raw.vcf  | vt normalize -r reference/ref.fa - | bcftools annotate --remove '^INFO/TYPE,^INFO/
DP,^INFO/RO,^INFO/AO,^INFO/AB,^FORMAT/GT,^FORMAT/DP,^FORMAT/RO,^FORMAT/AO,^FORMAT/QR,^FORMAT/QA,^FORMAT/GL' > snps.filt.vcf 2>> snps.log
normalize v0.5

options:     input VCF file                                  -
         [o] output VCF file                                 -
         [w] sorting window size                             10000
         [n] no fail on reference inconsistency for non SNPs false
         [q] quiet                                           false
         [d] debug                                           false
         [r] reference FASTA file                            reference/ref.fa

stats: biallelic
          no. left trimmed                      : 1300
          no. right trimmed                     : 22047
          no. left and right trimmed            : 1773
          no. right trimmed and left aligned    : 0
          no. left aligned                      : 0

       total no. biallelic normalized           : 25120

       multiallelic
          no. left trimmed                      : 0
          no. right trimmed                     : 0
          no. left and right trimmed            : 0
          no. right trimmed and left aligned    : 0
          no. left aligned                      : 0

       total no. multiallelic normalized        : 0

       total no. variants normalized            : 25120
       total no. variants observed              : 48598
       total no. reference observed             : 0

Time elapsed: 2.09s

[11:57:30] Running: snpEff ann -noLog -noStats -no-downstream -no-upstream -no-utr -c reference/snpeff.config -dataDir . ref snps.filt.vcf > snps.vcf 2>> snps.log
[11:57:42] Running: /isg/shared/apps/snippy/4.4.5/bin/snippy-vcf_to_tab --gff reference/ref.gff --ref reference/ref.fa --vcf snps.vcf > snps.tab 2>> snps.log
[11:57:53] Running: /isg/shared/apps/snippy/4.4.5/bin/snippy-vcf_extract_subs snps.filt.vcf > snps.subs.vcf 2>> snps.log
[11:57:55] Running: bcftools convert -Oz -o snps.vcf.gz snps.vcf 2>> snps.log
[11:57:56] Running: bcftools index -f snps.vcf.gz 2>> snps.log
[11:57:56] Running: bcftools consensus -f reference/ref.fa -o snps.consensus.fa snps.vcf.gz 2>> snps.log
[11:57:57] Error running command, check JKS2176_snippy_reads_chromosome/snps.log

When I check the log file it looks like a BCF tools issue?
[snps.log](https://github.com/tseemann/snippy/files/4104736/snps.log)

Any help getting around this would be great!

**Comment by tseemann** [comment]
Your log file says:
```
### bcftools consensus -f reference/ref.fa -o snps.consensus.fa snps.vcf.gz

Note: the --sample option not given, applying all records regardless of the genotype
The fasta sequence does not match the REF allele at c_000000000001:2450731:
   .vcf: [C]
   .vcf: [CG] <- (ALT)
   .fa:  [G]GGGCTCAGCCCAGGTTCTCGCCGTAGACGCCCTCCCGCCAGCGGTGCCTGGCGGGAGGGGGACGACCCCGGCCGCCGCCGATTCCGGCGGCGCGGTCGGGCGCGCGGCCGACCACCCCCGTCGTCGACGGCGCCACCGAGGCCCGCACCCCGCCCCGCGGTGTCCCGGTCA49||÷+

```

**Comment by sgl07007** [comment]
So do I need to add the --sample flag to the BCFtools step? This seems to be a known issue with BCFtools (https://www.biostars.org/p/354264/) but I am not sure what to do here.

**Comment by tseemann** [comment]
It is saying that your reference genome and your VCF file don't agree.
This is very strange.
Does it happen if you use a FASTA reference instead of a Genbank one?

**Comment by tseemann** [comment]
UPDATE: it seems this bug was fixed in bcftools 1.10.  I am releasing a new version of Snippy which requires this minimum version.
